src/lib/Libifl/pbsD_submit.c

```diff
@@ -201,10 +200,24 @@ __pbs_submit(int c, struct attropl  *attrib, char *script, char *destination, ch
 	for (pal = attrib; pal; pal = pal->next)
 		pal->op = SET;		/* force operator to SET */
 
+	cred_info = (struct cred_info *) ptr->th_cred_info;
+
+	if ((!script || (*script == '\0')) && (!cred_info || (cred_info->cred_len <= 0))) {
+		/* no cred and no script, let's request implicit commit to cut one message exchange */
+		if (!(extend = pbs_strcat(&extend, NULL, EXTEND_OPT_IMPLICIT_COMMIT))) {
```

Comment:
I think you missed freeing extend? or is it intentional because this is a client call?

Reply:
i will free it